src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -53,59 +53,56 @@ object PartyAddons {
             event.isCanceled = true
         } else if (partyStartPattern.matches(message)) {
             party.clear()
             event.isCanceled = true
-        } else if (message.startsWith("§eParty Leader")) {
-            val leader = leaderPattern.find(message)
-
-            if (leader != null) {
-                val name = leader.groups["name"]?.value!!
-                val status = leader.groups["status"]?.value!!
-                val rank = leader.groups["rank"]?.value!!
-
-                party.add(PartyMember(name, PartyMemberType.LEADER, status, rank))
-                event.isCanceled = true
-            }
-        } else if (message.startsWith("§eParty Moderators: ")) {
-            playerPattern.findAll(message.substringAfter("§eParty Moderators: ")).forEach {
-                val name = it.groups["name"]?.value!!
-                val status = it.groups["status"]?.value!!
-                val rank = it.groups["rank"]?.value!!
-
+        } else if (message.startsWith("§eParty Leader: ")) {
+            leaderPattern.find(message)?.destructured?.let { (rank, name, status) ->
                 party.add(
                     PartyMember(
                         name,
-                        PartyMemberType.MODERATOR,
+                        PartyMemberType.LEADER,
                         status,
                         rank
                     )
                 )
+                event.isCanceled = true
+            }
+        } else if (message.startsWith("§eParty Moderators: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Moderators: ")).forEach {
+                it.destructured.let { (rank, name, status) ->
+                    party.add(
+                        PartyMember(
+                            name,
+                            PartyMemberType.MODERATOR,
+                            status,
+                            rank
+                        )
+                    )
+                }
             }
             event.isCanceled = true
         } else if (message.startsWith("§eParty Members: ")) {
             playerPattern.findAll(message.substringAfter("§eParty Members: ")).forEach {
-                val name = it.groups["name"]?.value ?: return
-                val status = it.groups["status"]?.value ?: return
-                val rank = it.groups["rank"]?.value ?: return
-
-                party.add(
-                    PartyMember(
-                        name,
-                        PartyMemberType.MEMBER,
-                        status,
-                        rank
+                it.destructured.let { (rank, name, status) ->
+                    party.add(
+                        PartyMember(
+                            name,
+                            PartyMemberType.MEMBER,
+                            status,
+                            rank
+                        )
                     )
-                )
+                }
```

Comment:
I feel like all of this could be changed into a single block instead of 3 different cases.